tests/test_wav2vec2.py

```diff
@@ -42,36 +45,54 @@ def _get_batches(self):
         return batch, hf_batch, tf_labels, hf_labels
 
     @partial(requires_lib, lib=["torch", "transformers"])
-    def _test_inference(self, test_graph_mode=False):
+    def _test_inference(self, model_id, hf_model_id, test_graph_mode=False):
         @tf.function(autograph=True, jit_compile=True)
         def tf_forward(*args, **kwargs):
             return tf_model(*args, **kwargs)
 
         batch, hf_batch, _, _ = self._get_batches()
 
-        tf_model = Wav2Vec2ForCTC.from_pretrained(MODEL_ID, input_shape=batch.shape)
-        hf_model = HFWav2Vec2ForCTC.from_pretrained(HF_MODEL_ID)
+        tf_model = Wav2Vec2Model.from_pretrained(model_id, input_shape=batch.shape)
+        hf_model = HFWav2Vec2Model.from_pretrained(hf_model_id)
+
+        if tf_model.config.is_robust:
+            attention_mask = np.ones(batch.shape, dtype=np.int32)
+            attention_mask[0, -1000:] = attention_mask[1, -132:] = 0
+            hf_attention_mask = torch.from_numpy(attention_mask)
+            attention_mask = tf.convert_to_tensor(attention_mask)
+        else:
+            attention_mask = hf_attention_mask = None
 
         if test_graph_mode:
-            tf_out = tf_forward(batch, training=False)
+            tf_out = tf_forward(batch, attention_mask=attention_mask, training=False)
         else:
-            tf_out = tf_model(batch, training=False)
+            tf_out = tf_model(batch, attention_mask=attention_mask, training=False)
         with torch.no_grad():
-            hf_out = hf_model(hf_batch)
+            hf_out = hf_model(hf_batch, attention_mask=hf_attention_mask)
 
         tf_logits = tf_out.numpy()
-        hf_logits = hf_out["logits"].numpy()
+        hf_logits = hf_out["last_hidden_state"].numpy()
 
         assert tf_logits.shape == hf_logits.shape, "Oops, logits shape is not matching"
         assert np.allclose(
-            hf_logits, tf_logits, atol=0.004
+            hf_logits, tf_logits, atol=1e-3
         ), f"difference: {np.max(hf_logits - tf_logits)}"
 
     def test_inference(self):
-        self._test_inference(test_graph_mode=False)
+        model_id, hf_model_id = "vasudevgupta/gsoc-wav2vec2", "facebook/wav2vec2-base"
+        self._test_inference(model_id, hf_model_id, test_graph_mode=False)
+
+    def test_wav2vec2_robust(self):
+        model_id, hf_model_id = "vasudevgupta/gsoc-wav2vec2-robust", "facebook/wav2vec2-large-robust"
+        self._test_inference(model_id, hf_model_id, test_graph_mode=False)
+
+    def test_wav2vec2_xlsr(self):
+        model_id, hf_model_id = "vasudevgupta/gsoc-wav2vec2-xlsr-53", "facebook/wav2vec2-large-xlsr-53"
+        self._test_inference(model_id, hf_model_id, test_graph_mode=False)
```

Comment:
Hello @MorganR, @sayakpaul,

I have ported [facebook/wav2vec2-large-xlsr-53](https://huggingface.co/facebook/wav2vec2-large-xlsr-53) in TensorFlow as suggested by @MorganR last week. This test ensures that port is successful. If everything looks alright to you, I will add TFSavedModel for this checkpoint to TFHub. Shall I go for adding SavedModel to TFHub??